src/core/Typing.ml

```diff
@@ -117,43 +117,58 @@ let check_dim_scope oxs r =
     | _ -> ()
 
 let check_valid_cofibration ?xs:(xs = None) cofib =
```

Comment:
Is this the algorithm to check that bipartiteness condition? If so, could you mention that in the code?

Reply:
Does it look better now?